libbeat/beat/events/util.go

```diff
@@ -0,0 +1,70 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package events
+
+import "github.com/elastic/beats/v7/libbeat/beat"
+
+const (
+	// FieldMetaID defines the ID for the event. Also see FieldMetaOpType.
+	FieldMetaID = "_id"
+
+	// FieldMetaAlias defines the index alias to use for the event. If set, it takes
+	// precedence over values defined using FieldMetaIndex or FieldMetaRawIndex.
+	FieldMetaAlias = "alias"
+
+	// FieldMetaIndex defines the base index name to use for the event. The value is suffixed
+	// with a Y-m-d value based on the event's timestamp. If set, it takes precedence over the
+	// value defined using FieldMetaRawIndex.
+	FieldMetaIndex = "index"
+
+	// FieldMetaRawIndex defines the raw index name to use for the event. It is used as-is, without
+	// any additional manipulation.
+	FieldMetaRawIndex = "raw_index"
+
+	// FieldMetaPipeline defines the ingest node pipeline to use for this event.
+	FieldMetaPipeline = "pipeline"
+
+	// FieldMetaOpType defines the metadata key name for event operation type to use with the Elasticsearch
+	// Bulk API encoding of the event. The key's value can be an empty string, `create`, `index`, or `delete`.
+	// If empty, `create` will be used if FieldMetaID is set; otherwise `index` will be used.
+	FieldMetaOpType = "op_type"
+
+	FieldMetaOpTypeCreate MetaOpType = iota
+	FieldMetaOpTypeDelete
+	FieldMetaOpTypeIndex
```

Comment:
nit (grouping). Better use a separate `const ( ... )` for the enum to separate keys from the enum.
Maybe naming could just be `OpTypeCreate`, `OpTypeDelete`, `OpTypeIndex`.

Reply:
What happens if the field was not set or the configured opType was an empty string? Would it make sense to introduce `OpTypeDefault = 0` (given that we apply a version check in the ES output)?